tutorial/markdown/nodejs/quickstart-typescript/quickstart-typescript.md

```diff
@@ -0,0 +1,483 @@
+
+```ts
+    await makeResponse(res, async () => {
+        await airportCollection.remove(req.params.id)
+        res.status(204)
+        return
+    })
+```
+
+Delete Airport by Airport ID by using [`.remove()`](https://docs.couchbase.com/sdk-api/couchbase-node-client/classes/Collection.html#remove) method and returns a 404 if the document is not found.
+
+## List Airport
+
+This endpoint retrieves the list of airports in the database. The API has options to specify the page size for the results and country from which to fetch the airport documents.
+
+[SQL++](https://docs.couchbase.com/nodejs-sdk/current/howtos/n1ql-queries-with-sdk.html) is a powerful query language based on SQL, but designed for structured and flexible JSON documents. We will use a SQL++ query to search for airports with Limit, Offset, and Country option.
+
+Navigate to the `listAirport` method in the `airportController.ts` file. This endpoint is different from the others we have seen before because it makes the SQL++ query rather than a key-value operation. This usually means more overhead because the query engine is involved. For this query, we are using the predefined indices in the `travel-sample` bucket. We can create an additional [index](https://docs.couchbase.com/server/current/learn/services-and-indexes/indexes/indexing-and-query-perf.html) specific for this query to make it perform better.
+
+First, we need to get the values from the query string for country, limit, and Offset that we will use in our query. These are pulled from the `request.query` method.
+
+This end point has two queries depending on the value for the country parameter. If a country name is specified, we retrieve the airport documents for that specific country. If it is not specified, we retrieve the list of airports across all countries. The queries are slightly different for these two scenarios.
+
+We build our SQL++ query using the [named parameters](https://docs.couchbase.com/nodejs-sdk/current/howtos/n1ql-queries-with-sdk.html#queries-placeholders) specified by `$` symbol for both these scenarios. The difference between the two queries is the presence of the `country` parameter in the query. Normally for the queries with pagination, it is advised to order the results to maintain the order of results across multiple queries.
+
+Next, we pass that `query` and the `parameters` to the  [scope.query()](https://docs.couchbase.com/sdk-api/couchbase-node-client/classes/Scope.html#query) method and return the results.
+
+```ts
+    const country = (req.query.country as string) ?? ''
+    let limit = parseInt(req.query.limit as string, 10) || 10
+    let offset = parseInt(req.query.offset as string, 10) || 0
+    let query: string
+    type QueryOptions = {
+        parameters: {
+            COUNTRY?: string
+            LIMIT: number
+            OFFSET: number
+        }
+    }
+    let options: QueryOptions
+    if (country !== '') {
```

Comment:
Can you add a new line here just to separate these two parts?

Reply:
Added a new line